### PR TITLE
Use medusajs/medusa package to register the CLI entrypoint

### PR DESCRIPTION
**Pull request by thetutlage**
I see that multiple packages registers itself to the bin as `medusa` binary. I went ahead and made `@medusajs/medusa` to do it only. 

Let me know if my approach is wrong :)

### CODE REVIEW

File: packages/cli/medusa-cli/package.json [hunk 1]
```diff
@@ -3,9 +3,6 @@
   "version": "1.3.22",
   "description": "Command Line interface for Medusa Commerce",
   "main": "dist/index.js",
-  "bin": {
-    "medusa": "cli.js"
-  },
```
Comment (adrien2p) [inline packages/cli/medusa-cli/package.json]: are you sure this can be removed?

File: packages/medusa/package.json [hunk 2]
```diff
@@ -3,7 +3,9 @@
   "version": "1.20.5",
   "description": "Building blocks for digital commerce",
   "main": "dist/index.js",
-  "bin": "./cli.js",
```
Comment (thetutlage) [inline packages/medusa/package.json:6]: Then should we remove this @adrien2p ?

File: packages/cli/medusa-cli/package.json (hunk 1 above)
Comment (thetutlage) [reply packages/cli/medusa-cli/package.json]: Added back

File: packages/medusa/package.json (hunk 2 above)
Comment (thetutlage) [reply packages/medusa/package.json:6]: Removed

**Review by sradevski — APPROVED** [review]

**Comment by thetutlage** [comment]
@adrien2p Want to take a final look?

**Comment by olivermrbl** [comment]
@thetutlage should we test this with a snapshot? It seems the pipeline is failing 🤔

**Review by adrien2p — APPROVED** [review]

**Pull request merged**